### PR TITLE
Create a YouTube Video Downloader using Python

### LINKED ISSUE #6324: Create a Youtube video Dowloader with Python
### Introduction
YouTube is a well-known video-sharing platform. It takes a long time to download a YouTube video or playlist. When you use Downloader to download a video or try to get it from a random website, you run the danger of your personal information being stolen. This task is incredibly straightforward, efficient, and safe when using the Python Tkinter module which gets the video directly from youtube for you.

### Key takeaways
- Prerequisites
- Requirements and Installations
- Developing the layout/ the GUI
- Connecting the GUI to Youtube using PyTube
- Testing the GUI workability

#### Article quality
This article will be easy to follow for those aspiring to use Python for Applications development. I believe knowledge can never be exhausted and this article will be novice to be clear even to Python beginners.

#### References
N/A.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by vifabulous**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6324 
> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Plagiarism check complete

**Comment by lalith1403** [comment]
Kindly provide your author folder name.

File: content/articles/youtube-video-downloader-using-python/index.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+The download button allows you to download the video by calling the download function in the next step. It only takes one line of code, as shown in the code block below.
+
+```python
+Button(root, text='Download', font='san-serif 16 bold', bg='red', padx=2,command="download").place(x=100, y=150)
+ 
+```
+The command `download` is the attribute that initiates the download process when the button is pressed. It is called from the download function created below for getting the video from YouTube.
+
+### Creating the download function
+Up to this point, you have made an API where you can paste your link from youtube, but the download button is not yet functional. To fetch the video using the URL, you will have to create a download function called the download button.
```
Comment (lalith1403) [inline content/articles/youtube-video-downloader-using-python/index.md:114]: This is not an API though. Please check.

**Comment by lalith1403** [comment]
Well written article. Easy to read and comprehend. Please resolve the above issue.

**Comment by vifabulous** [comment]
> Kindly provide your author folder name.

it is named `kamau victor`

**Comment by vifabulous** [comment]
@lalith1403 I have resolved the stated issue

**Comment by vifabulous** [comment]
@lalith1403 

> Well written article. Easy to read and comprehend. Please resolve the above issue.

@lalith1403 I have resolved the issue, was there any other issue with the article?

**Comment by hectorkambow** [comment]
@vifabulous 

Question: You had just mentioned that you would not be using this github profile? 

Instead you mentioned you would be using @del-ui

**Comment by vifabulous** [comment]
At that time this PR was already under review

**Comment by vifabulous** [comment]
I will shift it to the other account

**Pull request closed without merging**